src/main/java/com/amazonaws/encryptionsdk/caching/CachingCryptoMaterialsManager.java

```diff
@@ -100,13 +101,33 @@ public Builder withBackingMaterialsManager(CryptoMaterialsManager backingCMM) {
          * This method is equivalent to calling {@link #withBackingMaterialsManager(CryptoMaterialsManager)} passing a
          * {@link DefaultCryptoMaterialsManager} constructed using your {@link MasterKeyProvider}.
          *
+         * @deprecated {@link MasterKeyProvider}s have been deprecated in favor of {@link Keyring}s.
+         *
          * @param mkp The MasterKeyProvider to invoke on cache misses
          * @return this builder
          */
+        @Deprecated
         public Builder withMasterKeyProvider(MasterKeyProvider mkp) {
             return withBackingMaterialsManager(new DefaultCryptoMaterialsManager(mkp));
         }
 
+        /**
+         * Sets the {@link Keyring} that should be queried when the {@link CachingCryptoMaterialsManager}
+         * incurs a cache miss.
+         *
+         * You can set either a Keyring or a CryptoMaterialsManager to back the CCMM - the last value set will
+         * be used.
```

Comment:
This is a good callout. I didn't like it at first until I realized all builders are like this (silently overwriting existing parameters if you call them multiple times).

Can you specify that all three of the builder methods overwrite each other, on all three methods? That will make this sit better.